common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsEvent.kt

```diff
@@ -0,0 +1,88 @@
+package com.revenuecat.purchases.common.diagnostics
+
+import org.json.JSONArray
+import org.json.JSONObject
+
+sealed class DiagnosticsEvent(val diagnosticType: String) {
+    companion object {
+        private const val VERSION_KEY = "version"
+        private const val TYPE_KEY = "type"
+
+        private const val VERSION = 1
+    }
+
+    data class Log(
```

Comment:
This type of event is not yet supported in the backend but I believe we will likely need it. We can remove it later if not.